manager/tests/startup_tests.py

```diff
@@ -0,0 +1,145 @@
+                        expected_events.pop(0)
+            except queue.Empty:
+                return False
+
+        return True
+
+    def test(self, charging_session_cmd: dict, expected_events: list[str], timeout: float) -> bool:
+        end_of_time = time.time() + timeout
+
+        if not self._ready_event.wait(timeout):
+            
+            return False
+
+        # fetch fulfillment
+        car_sim_ff = self._setup.connections['test_control'][0]
+
+        # enable simulator    
+        self._mod.call_command(car_sim_ff, 'enable', {'value': True})
+
+        # start charging simulation
+        logging.info(charging_session_cmd['value'])
+        self._mod.call_command(car_sim_ff, 'executeChargingSession', charging_session_cmd)
+
+        return self._pool_for_expected_events(expected_events, end_of_time)
+
+@pytest.mark.everest_core_config('config-sil.yaml')
+@pytest.mark.asyncio
+async def test_001_start_test_module(everest_core: EverestCore):
+    logging.info(">>>>>>>>> test_001_start_test_module <<<<<<<<<")
+
+    test_connections = {
+        'test_control': [Requirement('car_simulator', 'main')],
+        'connector_1': [Requirement('connector_1', 'evse')]
+    }
+
+    everest_core.start(standalone_module='probe', test_connections=test_connections)
+    logging.info("everest-core ready, waiting for probe module")
+
+    session = RuntimeSession(str(everest_core.prefix_path), str(everest_core.everest_config_path))
+
+    probe = ProbeModule(session)
+
+    if everest_core.status_listener.wait_for_status(10, ["ALL_MODULES_STARTED"]):
+        everest_core.all_modules_started_event.set()
+        logging.info("set all modules started event...")
+    charging_session_cmd = {'value': 'sleep 1;iec_wait_pwr_ready;sleep 1;draw_power_regulated 16,3;sleep 5'}
+    expected_events = ['TransactionStarted', 'ChargingStarted']
+    
+    assert probe.test(charging_session_cmd,expected_events,20)
+
+@pytest.mark.asyncio
+async def test_000_demo_run(everest_core: EverestCore):
+    logging.info(">>>>>>>>> test_000_demo_run <<<<<<<<<")
+
+    test_connections = {
+        'test_control': [Requirement('car_simulator', 'main')],
+        'connector_1': [Requirement('connector_1', 'evse')]
+    }
+
+    everest_core.start(standalone_module='probe',test_connections=test_connections)
+
+    logging.info("everest-core ready, waiting for probe module")
+
+    session = RuntimeSession(str(everest_core.prefix_path), str(everest_core.everest_config_path))
+
+    probe = ProbeModule(session)
+
+    if everest_core.status_listener.wait_for_status(10, ["ALL_MODULES_STARTED"]):
+        everest_core.all_modules_started_event.set()
+        logging.info("set all modules started event...")
+
+    charging_session_cmd = {'value': 'sleep 1;iec_wait_pwr_ready;sleep 1;draw_power_regulated 16,3;sleep 5;pause;sleep 5;diode_fail;sleep 5'}
+    expected_events = ['TransactionStarted', 'ChargingStarted','ChargingPausedEV', {'error': 'CarDiodeFault'}]
+    
```

Comment:
Again, for my understanding, the main change here, over the `startup_tests.py` in `everest-core` is that `everest-core` has a set of hardcoded commands. Here, after initializing the modules, we also send the sequence of commands and expect a sequence of events (with the corresponding changes to `ProbeModule`, of course).

Reply:
The biggest change is that we were only getting high level events back. I updated the array to pass back either the event name or a dict with more details. This allows for deeper assertions in the test. There is way more that could probably be done but this proved we could inspect more using the stand alone probe module.